ibis/backends/tests/test_client.py

```diff
@@ -1460,3 +1469,37 @@ def test_list_databases_schemas(con_create_database_schema):
             con_create_database_schema.drop_schema(schema, database=database)
     finally:
         con_create_database_schema.drop_database(database)
+
+
+@pytest.mark.notyet(
+    ["pandas", "dask", "polars", "datafusion"],
+    raises=NotImplementedError,
+    reason="In process backends have nothing to close",
+)
+@pytest.mark.notyet(
+    ["trino", "clickhouse", "impala", "bigquery"],
+    reason="Backend client does not conform to DB-API, subsequent op does not raise",
+)
+def test_close_connection(con):
+    new_con = getattr(ibis, con.name).connect(*con._con_args, **con._con_kwargs)
+
+    # Run any command that hits the backend
+    _ = new_con.list_tables()
+    new_con.disconnect()
+
+    # DB-API states that subsequent execution attempt should raise
+    with pytest.raises(
+        (
+            DuckDBConnectionException,
+            ExaRuntimeError,
+            MySQLInterfaceError,
+            OracleInterfaceError,
+            PsycoPg2InterfaceError,
+            Py4JJavaError,
+            PyDruidError,
+            PyODBCProgrammingError,
+            SnowflakeDatabaseError,
+            sqlite3ProgrammingError,
+        )
+    ):
```

Comment:
I know `pytest.raises(Exception)` is considered evil, but writing all of these out is running afoul of our (good) test environment isolation.  Hmmmmm

Reply:
Using `pytest.raises(Exception)` seems fine to me honestly. We only care that things don't work, not necessarily why.